code/game/objects/pool/pool.dm

```diff
@@ -0,0 +1,308 @@
+
+/obj/machinery/pool_filter/Initialize()
+	. = ..()
+	create_reagents(100, OPENCONTAINER) //If you're a terrible terrible clown and want to dump reagents into the pool.
+	if(preset_reagent_type)
+		reagents.add_reagent(preset_reagent_type, 100)
+	var/area/AR = get_area(src)
+	for(var/turf/open/indestructible/sound/pool/water in get_area_turfs(AR))
+		if(!istype(water))
+			continue //Only affect pool turfs.
+		if(id && water.id != id)
+			continue //Not the same id. Fine. Ignore that one then!
+		pool += water
+	GLOB.pool_filters += src
+
+//Brick can set the pool to low temperatures remotely. This will probably be hell on malf!
+
+/obj/machinery/pool_filter/attack_robot(mob/user)
+	. = ..()
+	wrench_act(user, null)
+
+/obj/machinery/pool_filter/attack_ai(mob/user)
+	. = ..()
+	wrench_act(user, null)
+
+/obj/machinery/pool_filter/wrench_act(mob/living/user, obj/item/I)
+	. = ..()
+	var/newTemp = input(user, "Set a new temperature for [src] (Kelvin).", "[src]", null) as num
+	if(!newTemp)
+		return
+	newTemp = CLAMP(newTemp, T0C, 320)
+	desired_temperature = newTemp
+	return FALSE
+
+/obj/machinery/pool_filter/process()
+	if(!pool?.len)
+		return //No use having one of these processing for no reason is there?
+	if(!is_operational())
+		return
+	use_power(idle_power_usage)
+	var/delta = (current_temperature > desired_temperature) ? -0.5 : 0.5
+	current_temperature += delta
+	current_temperature = CLAMP(current_temperature, T0C, desired_temperature)
+	var/trans_amount = reagents.total_volume / pool.len //Split up the reagents equally.
+	for(var/turf/open/indestructible/sound/pool/water in pool)
+		if(reagents.reagent_list.len)
+			water.set_colour(mix_color_from_reagents(reagents.reagent_list))
+		else
+			water.set_colour("#009999")
+		if(water.contents.len && reagents.total_volume > 0)
+			for(var/mob/living/M in water)
+				if(!istype(M))
+					continue
+				var/datum/reagents/splash_holder = new/datum/reagents(trans_amount) //Take some of our reagents out, react them with the pool denizens.
+				splash_holder.my_atom = water
+				reagents.trans_to(splash_holder, trans_amount, transfered_by = src)
+				splash_holder.chem_temp = current_temperature
+				if(prob(80))
+					splash_holder.reaction(M, TOUCH)
+				else //Sometimes the water penetrates a lil deeper than just a splosh.
+					splash_holder.reaction(M, INGEST)
+				splash_holder.trans_to(M, trans_amount, transfered_by = src)	//Actually put reagents in the mob
+				qdel(splash_holder)
+				var/mob/living/carbon/C = M
+				if(current_temperature <= 283.5) //Colder than 10 degrees is going to make you very cold
+					if(iscarbon(M))
+						C.adjust_bodytemperature(-80, 80)
+					to_chat(M, "<span class='warning'>The water is freezing cold!</span>")
+				else if(current_temperature >= 308.5) //Hotter than 35 celsius is going to make you burn up
+					if(iscarbon(M))
+						C.adjust_bodytemperature(35, 0, 500)
+					M.adjustFireLoss(5)
```

Comment:
Wouldn't this be done by adjusting the bodytemp to be hotter anyway and just make the pool insanely dangerous at high temps?